### ISSUE TITLE
[CoreELEC] Kodi still run the old ISA binary after ISA addon update

### ISSUE THREAD

**Issue by CastagnaIT** [peak3d/inputstream.adaptive#727]
i was testing to fix a problem,
but i found another strange thing, if someone want to try check this ISA behaviour:
- run a DRM video (at this point to me raised an error for other reasons)
- update ISA
- try run another DRM video, kodi still run the old ISA binary instead use the new one

by using CoreELEC on ARM device

I don't know if it happened because it went wrong the first time,
only after device reboot i was able to run a DRM video with the updated ISA

**Comment by vascobraga41** [comment]
We've discussed this in CE and wonder if it's caused by only using `dlopen()` and and never `dlclose()` when handling libwidevine. Can the previous load stay in the memory?

**Comment by CastagnaIT** [comment]
are you saying that dlopen affects the release of ISA binary?

**Comment by vpeter4** [comment]
Inputstream Adaptive addon loads the libwidevinecdm.so library with dlopen() call. But dlclose() is not used. Which means that even the next dlopen() call is trying to load new version of the library the old remains in memory and still used?

**Comment by CastagnaIT** [comment]
good question...
but here i have open the Issue because i encoutered similar problem but with ISA addon binary

however if you enable dlclose in ISA
https://github.com/xbmc/inputstream.adaptive/blob/58c54c7b641f69a862ed2fa5a640b5fa2db9c711/wvdecrypter/cdm/base/native_library_posix.cc#L41

kodi will crash when unload it (so when you stop a video), if you want try:
[isa-linux-armv8-matrix-dlclose.zip](https://github.com/xbmc/inputstream.adaptive/files/6748035/isa-linux-armv8-matrix-dlclose.zip)

I think that's why it was commented on

**Comment by glennguy** [comment]
Sounds similar to #454 ? That issue seems to be related to the symlinks that are present in CE/LE

**Comment by vpeter4** [comment]
> kodi will crash when unload it (so when you stop a video)

Yes, you are right. I blame the library for that :)

**Comment by matthuisman** [comment]
Is this on Kodi 19 or 18?
Pretty sure it is same issue as @glennguy linked 
Which is a Kodi / binary addon issue, not a IA issue.

**Comment by CastagnaIT** [comment]
yes is referred to the same problem of the linked Issue
i didn't realise,
but similar thing happen to the loaded libwidevinecdm.so but should be opened an appropriate new Issue,
a simply test just remove folder "widevine" folder in "cdm" folder and widevine will not works anymore until kodi reboot
where dlclose should fix it

**Comment by matthuisman** [comment]
There is also the ssd_wv that gets opened too. Which that then opens the widevine blob. Maybe that isn't closed?
